source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -120,6 +121,28 @@ AWSLambdaFilter::decodeHeaders(Http::RequestHeaderMap &headers,
   return Http::FilterHeadersStatus::StopIteration;
 }
 
+
+Http::FilterHeadersStatus 
+AWSLambdaFilter::encodeHeaders(Http::ResponseHeaderMap &headers, bool ) {
+
+  if (!headers.get(AWSLambdaHeaderNames::get().FunctionError).empty()){
+    // We treat upstream errors as if it was any other upstream error
+    headers.setStatus(504);
+    return Http::FilterHeadersStatus::Continue;
+  }
+  return Http::FilterHeadersStatus::StopIteration;
+}
+
+Http::FilterDataStatus AWSLambdaFilter::encodeData(Buffer::Instance &,
+                                                        bool ) {
+  return Http::FilterDataStatus::Continue;
+}
+
+Http::FilterTrailersStatus
+AWSLambdaFilter::encodeTrailers(Http::ResponseTrailerMap &) {
+  return Http::FilterTrailersStatus::StopIteration;
+}
```

Comment:
why not move these to the header file like we did with `encode100ContinueHeaders`?

Reply:
Yah makes sense. Do we have a general guideline for complexity allowed in header? Something like if it is anything more than a no-op then dont put it in header?

Namely I am even wondering if even the encode headers should go in to the header file.
My gut feeling is no but while we are having the discussion its good know.

Reply:
We have generally been following upstream envoy for conventions.

I was just calling out that we were being inconsistent with our approach here, but i don't have a super strong opinion either way